src/sentry/incidents/endpoints/project_alert_rule_details.py

```diff
@@ -31,11 +35,8 @@ def put(self, request, project, alert_rule):
             data=data,
             partial=True,
         )
-
         if serializer.is_valid():
-            try:
-                alert_rule = serializer.save()
-            except ChannelLookupTimeoutError:
+            if alert_rule_data_requires_async_lookup(project.organization, request.user, data):
```

Comment:
@iProgramStuff Won't `alert_rule_data_requires_async_lookup` raise a `ChannelLookupTimeoutError` error here if we need to do the async lookup?

Reply:
We don't do the lookup in this function if the flag is False. The task will do the lookup and then update redis with `failed` if we get `ChannelLookupTimeoutError`